lib/galaxy/tools/actions/__init__.py

```diff
@@ -903,13 +903,21 @@ def _wrapped_params(
         wrapped_params = WrappedParameters(trans, tool, incoming, input_datasets=input_datasets)
         return wrapped_params
 
-    def _get_on_text(self, inp_data):
+    def _get_on_text(self, inp_data, inp_dataset_collections):
         input_names = []
-        for data in reversed(list(inp_data.values())):
+        collection_names = []
+        for collections in inp_dataset_collections.values():
+            for dataset_collection, _ in collections:
+                if getattr(dataset_collection, "hid", None):
+                    collection_names.append(str(dataset_collection.hid))
+
+        for input_name in reversed(inp_data):
+            data = inp_data[input_name]
+            if input_name in inp_dataset_collections:
+                continue
             if getattr(data, "hid", None):
-                input_names.append(f"data {data.hid}")
-
-        return on_text_for_names(input_names)
+                input_names.append(f"{data.hid}")
```

Comment:
That's now just the hid, so I would maybe rename `input_names` and `collection_names` to avoid the confusion, and skip the string casting as well.

Reply:
You mean change the function signature ? Yes, I think that would be good.